src/lib/peer.spec.ts

```diff
@@ -0,0 +1,91 @@
+import { Profile } from '../modules/profile/types'
+import { PeerAPI } from './peer'
+
+const peerApi = new PeerAPI('aPeerURL')
+
+describe('peerAPI', () => {
+  let address: string
+  let profile: Profile
+  beforeEach(() => {
+    address = 'anAddress'
+    profile = {
+      avatars: [
+        {
+          avatar: {
+            bodyShape: ''
+          },
+          description: 'aDescription',
+          name: 'aName'
+        }
+      ]
+    } as Profile
+  })
+
+  describe('when fetching the user profile', () => {
+    describe('and the useCache parameter is passed', () => {
+      describe('and the result is in the cache', () => {
+        beforeEach(() => {
+          peerApi.cache = { [address]: Promise.resolve(profile) }
+        })
+        it('should get the result from the cache', async () => {
+          expect(await peerApi.fetchProfile(address, { useCache: true })).toBe(
+            profile
+          )
+        })
+      })
+
+      describe('and the result is not in the cache', () => {
+        let profileUpdated: Profile
+        beforeEach(() => {
+          profileUpdated = {
+            avatars: [
+              {
+                avatar: {
+                  bodyShape: ''
+                },
+                description: 'aDescription',
+                name: 'aNewName'
+              }
+            ]
+          } as Profile
+          peerApi.cache = {}
+          jest
+            .spyOn(peerApi.lambdasClient, 'fetchProfiles')
+            .mockImplementation(() => Promise.resolve([profileUpdated]))
+        })
+        it('should fetch the profile', async () => {
+          expect(await peerApi.fetchProfile(address, { useCache: true })).toBe(
+            profileUpdated
+          )
```

Comment:
Is it possible to check that the cache was updated as well? Like, for example, we might be able to perform an:
```
await expect(peerApi.cache).resolves.toEqual(profileUpdated)
```
What if we add something like this to ensure the cache to be set?
The same applies for the case when the `useCache` parameter is not passed.

Reply:
added! in the case the `useCache` parameter is not used, the cache is set anyways, so the `expect` is the same in both cases.